<!DOCTYPE html>

<head>
	<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
	<title>Bio-DSL Manual</title>
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/default.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
</head>

<body>
<h1>Bio-DSL</h1>

<p>Bio-DSL is a domain-specific language for Bioinformatics domain. It has a simple syntax and a minimal set of keywords. The syntax has similarities to pythonâ€™s syntax and indenting. Biologists can write a workflow script using the visual elements offered in Bio-DSL web interface.</p>  

<h3><a name="keywords">Keywords:</a></h3>
The keywords of Bio-DSL are listed below:

<table>
  <tr>
    <th>Construct</th>
    <th>Keywords</th>
    <th>Example</th>
  </tr>
  <tr>
    <td>Conditional</td><td><font color="green">if</font></td><td><pre><code class="python">
for f in GetFiles('/public/MiSeq_SOP'):
    if GetDataType(f) == 'fastq':
        fq_html = fastqc.CheckQuality(f)
</code></pre></td>
  </tr>

  <tr>
     <td>Iteration</td> <td><font color="green">for ... in </font></td> <td><pre><code class="python">
for f in GetFiles('/public/MiSeq_SOP'):
    if GetDataType(f) == 'fastq':
        fq_html = fastqc.CheckQuality(f)
</code></pre>
</td>
  </tr>
  <tr>
  	<td>Parallelization</td>
  	<td><font color="green">parallel ... with</font></td><td><pre><code class="python">
datas = ['/public/MiSeq_SOP/F3D6_S194_L001_R1_001.fastq', '/public/MiSeq_SOP/F3D6_S194_L001_R2_001.fastq']
parallel:
    fq_html = fastqc.CheckQuality(datas[0])
with:
    fq_html = fastqc.CheckQuality(datas[1])</code></pre></td>
  </tr>
  <tr>
  	<td>Subworkflow</td><td><font color="green">task</font></td><td><pre><code class="python">
task AlignSequences(ref, data, data2):
    CheckQuality(data)
    CheckQuality(data2)

    data = pear.Merge(data, data2)
    data = bwa.Align(ref, data)
    data = SamToBam(data)
    return data

AlignSequences('/public/genomes/Chr1.cdna', '/public/MiSeq_SOP/F3D6_S194_L001_R1_001.fastq', '/public/MiSeq_SOP/F3D6_S194_L001_R2_001.fastq')
</code></pre></td>
  </tr>
  <tr>
        <td/><td/><td>An anonymous task without name is called at once. Named tasks are called from somewhere else.</td>
    </tr>
</table>


<h3><a name="syntax">Syntax:</a></h3>
Bio-DSL uses simple pythonic syntax and indenting. Advanced syntaxes like lambda, class, iterators, generators, annotation are not allowed. Here is a complete example of a Bio-DSL script:
<p>Like Python, Bio-DSL is structured with <a href="">indentation</a> i.e. the statements within a block line up vertically. The block ends at a line less indented or the end of the file. If a block has to be more deeply nested, it is simply indented further to the right.</p>
Here is a Bio-DSL indenting example:
<pre><code class="python">
task AlignSequences(ref, data, data2):
    CheckQuality(data)
    CheckQuality(data2)

    data = pear.Merge(data, data2)
    data = bwa.Align(ref, data)
    data = SamToBam(data)
    return data

AlignSequences('/public/genomes/Chr1.cdna', '/public/MiSeq_SOP/F3D6_S194_L001_R1_001.fastq', '/public/MiSeq_SOP/F3D6_S194_L001_R2_001.fastq')
</code></pre>

<p>Bio-DSL is a dynamic language. The type of a literal is inferred from the value. For example:</p>
<table>
<tr>
<td>Integer </td>
<td>i = 10</td>
</tr>
<tr>
<td>Float</td>
<td>f = 10.0f</td>
</tr>
<tr>
<td>Text</td>
<td>s = "Hello World!"</td>
</tr>
</table>


<h3>The Bio-DSL <a name="webui">web interface</a></h3>
The domain experts can use Bio-DSL web interface to quickly create a workflow. The IDE looks as below:
<img alt>
<p>
A typical set of events are:
<ul>
	<li>Search the desired service from the Services panel to the right.</li>
	<li>Double click the service to insert into the code editor.</li>
	<li>Alternatively, you can also start typing the service name on code editor and press Ctrl+Space for code completion.</li>
	<li>Search the data item from the DataSources panel to the left.</li>
	<li>Replace the data argument on the code editor by double clicking the desired data source on this panel.</li>
	<li>Run the workflow from command panel.</li>
	<li>An item is added to the left-bottom panel for the execution job. Check it for automatic status update of each step or double click for instant load.</li>
</ul>
</p>

<h3><a name="datasources">Data Sources Panel</a></h3>
<p>
This is the left-top panel of the user interface. It lists all the data items available from different file systems (posix, galaxy, hdfs). The public folder of each file system is available to all users for read access. Another folder named as username is available for read-write accesses. The generated files during workflow execution are usually found in this folder.
</p>
<p>To insert a data item into the code editor, first remove the argument name from the service call and then click right button on the item and select "To Editor".</p>

<h4>Data Source Toolbar</h4>
There is a small toolbar on top of data source panel.
<ul>
	<li>Add: Adds a folder at the selected item.</li>
	<li>Remove: Removes the selected item.</li>
	<li>Rename: Renames the selected item.</li>
	<li>Upload: Uploads an item from the client machine to the selected folder.</li>
	<li>Download: Downloads the selected file.</li>
	<li>Reload: Reloads the tree from the file system.</li>
	<li>About: Opens the help page.</li>
</ul>

<p>Sometimes, you may need to insert the selected data item in the arguments box. Right click on the selected item and press "To args" context menu item.</p>

<h3><a name="services">Services Panel</a></h3>
The services/tools for the workflows are listed on the right-top corner of the UI. There are 3 different access modes for the service.
<ol>
	<li>Public: Accessible by all users.</li>
	<li>Shared: Accessible only by those users who are given accesses.</li>
	<li>Private: Accessible only by the logged on user himself.</li>
</ol>
<p>There are two operation modes of the services shown by a dropdown list. If basic is selected, some advanced operations for experienced users are hidden from the service list.</p>

<h4><a name="addservice">Add New Service</a></h4>
<p>Experienced users can extend the capabilities of the system by adding new services to it. They can write a new python module or call external program from python code.</p>
In the service pane, the "+" button opens the dialog for Service Mapping. There is a coding tab to write the service code and a JSON tab to map the python code with DSL vocabulary. Here is a lean and thin service code:

<pre><code class="python">
def run_myservice(context, *args, *kwargs):
	return args[0] + args[1]
</code>
</pre>

Here is a typical json mapper of the above service. Only one of name or internal is mandatory.
<pre><code class="json">
{
	"functions": [
		{
			"org": "srlab",
			"package": "mylib",
			"module": "app.biowl.libraries.users.mainul.mylib",
			"group": "Arithmetic",
			"name": "MyService",
			"internal": "run_myservice",
			"desc": "Adds two numbers and returns the sum.",
			"returns": "number",
			"example": "data = mylib.MyService(x, y)",
			"params": [
				{
					"name": "x",
					"type": "number",
				},
				{
					"name": "y",
					"type": "number"
				}
			]
		}
	]
}
</code>
</pre>

The new service is by default private. If can be shared to specific users or to all.

<h3><a name="workflows">Workflows Panel</a></h3>
<p>The right-bottom panel lists the saved workflows of the system. Workflows can be public, private or shared. New workflows can be saved by clicking the "+" button.</p>

<h3><a name="jobs">Job Histories Panel</a></h3>
<p>The left-bottom panel lists all the workflow instances run by this user. You can stop currently running workflows. </p>
<p>Double click an item to display the status information of workflow execution.</p>
<p>If you check a single running job, the incremental execution status of each step is shown on the report viewer.</p>

</body>
